lib/sinon/behavior.js

```diff
@@ -126,6 +127,8 @@ var proto = {
             return args[this.returnArgAt];
         } else if (this.returnThis) {
             return context;
+        } else if (typeof this.throwArgAt === "number") {
+            throw args[this.throwArgAt];
```

Comment:
I think it should throw an exception. I know, it sounds ironic in this context, but an exception with a more telling message and not throwing `undefined` makes sense, I think. We do something similar in `yield`.